tests/Mono.Android-Tests/jni/reuse-threads.c

```diff
@@ -200,3 +206,115 @@ rt_invoke_callback_on_new_thread (CB cb)
 	return 0;
 }
 
+/* We return -2 for errors, because -1 is reserved for the pthreads PTHREAD_CANCELED special value, indicating that the
+ * thread was canceled. */
+static int
+_register_type_from_new_thread (void *data)
+{
+	RegisterFromThreadContext *context = (RegisterFromThreadContext*)data;
+
+	if (context == NULL) {
+		return -100;
+	}
+
+	JNIEnv *env = _get_env ("_register_type_from_new_thread");
+
+	if ((*env)->PushLocalFrame (env, 3) < 0) {
+		__android_log_print (ANDROID_LOG_INFO, "XA/RuntimeTest", "FAILURE: unable to create a local reference frame!");
+
+		if ((*env)->ExceptionOccurred (env)) {
+			(*env)->ExceptionClear (env);
+			(*env)->ExceptionDescribe (env);
+		}
+
+		return -101;
+	}
+
+	int ret                  = 0;
+	jclass ClassLoader_class = (*env)->FindClass (env, "java/lang/ClassLoader");
+	if (ClassLoader_class == NULL) {
+		ret = -102;
+		__android_log_print (ANDROID_LOG_INFO, "XA/RuntimeTest", "FAILURE: unable to find the 'java/lang/ClassLoader' class!");
+		goto cleanup;
+	}
+
+	jmethodID loadClass = (*env)->GetMethodID (env, ClassLoader_class, "loadClass", "(Ljava/lang/String;)Ljava/lang/Class;");
+	if (loadClass == NULL) {
+		ret = -103;
+		__android_log_print (ANDROID_LOG_INFO, "XA/RuntimeTest", "FAILURE: unable to get id of method 'loadClass' in the 'java/lang/ClassLoader' class!");
+		goto cleanup;
+	}
+
+	jobject loaded_class = (*env)->CallObjectMethod (env, context->class_loader, loadClass, context->java_type_name);
+
+	if ((*env)->ExceptionOccurred (env) != NULL) {
+		(*env)->ExceptionClear (env);
+		__android_log_print (ANDROID_LOG_INFO, "XA/RuntimeTest", "FAILURE: class '%s' cannot be loaded, Java exception thrown!", context->java_type_name);
+		(*env)->ExceptionDescribe (env);
```

Comment:
Calling `ExceptionDescribe()` needs to come *before* calling `ExceptionClear()`, as once you call `ExceptionClear()` there is no longer an exception to describe.